homeassistant/components/schedule/__init__.py

```diff
@@ -0,0 +1,334 @@
+"""
+Schedules temperature updates to components.
+
+Example component configuration:
+
+schedule:
+  entities:
+    -climate.home
+"""
+import json
+import logging
+
+import voluptuous as vol
+
+from homeassistant.helpers.event import async_track_state_change
+from homeassistant.util import dt
+from homeassistant.helpers.event import (
+    async_track_time_change)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_ENTITIES
+from homeassistant.core import callback, split_entity_id
+from homeassistant.helpers.storage import Store
+from homeassistant.components import websocket_api
+
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "schedule"
+
+STORAGE_VERSION = 1
+STORAGE_KEY = 'schedule'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_ENTITIES): cv.comp_entity_ids,
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+WS_TYPE_SCHEDULE_RULES = 'schedule/rules'
+SCHEMA_WEBSOCKET_GET_RULES = \
+    websocket_api.BASE_COMMAND_MESSAGE_SCHEMA.extend({
+        'type': WS_TYPE_SCHEDULE_RULES
+    })
+
+WS_TYPE_SCHEDULE_ENTITIES = 'schedule/entities'
+SCHEMA_WEBSOCKET_GET_ENTITIES = \
+    websocket_api.BASE_COMMAND_MESSAGE_SCHEMA.extend({
+        'type': WS_TYPE_SCHEDULE_ENTITIES
+    })
+
+WS_TYPE_SCHEDULE_CLEAR = 'schedule/clear'
+SCHEMA_WEBSOCKET_CLEAR = \
+    websocket_api.BASE_COMMAND_MESSAGE_SCHEMA.extend({
+        'type': WS_TYPE_SCHEDULE_CLEAR
+    })
+
+WS_TYPE_SCHEDULE_SAVE = 'schedule/save'
+SCHEMA_WEBSOCKET_SAVE = \
+    websocket_api.BASE_COMMAND_MESSAGE_SCHEMA.extend({
+        'type': WS_TYPE_SCHEDULE_SAVE,
+        'rules': list
+    })
+
+# Valid domains for entities with a schedule
+SCHEDULE_VALID_DOMAINS = ['climate', 'switch', 'light', 'input_boolean']
```

Comment:
What about `cover`? It would be useful if we could open a roller shutter at 06:00 AM or set the position to 40%.
Please consider that.
BTW Awesome idea and work. hopefully, a Scheduler based on your code will get added someday.